handlers/issue.go

```diff
@@ -1,71 +1,80 @@
+//function to handle issue comments
+func (s *Server) handleIssueCommentEvent(body []byte, client *github.Client) {
+	var commentEvent github.IssueCommentEvent
 
-	var prc github.IssueCommentEvent
-	err := json.Unmarshal(body, &prc)
+	err := json.Unmarshal(body, &commentEvent)
 	if err != nil {
 		glog.Errorf("fail to unmarshal: %v", err)
 	}
-	glog.Infof("prc: %v", prc)
-	comment := *prc.Comment.Body
+	//Get the owner info from the req body, and use this owner for the all api's
+	owner := *commentEvent.Repo.Owner.Login
+	ctx := context.Background()
+	//extract the user from login to validate whether user is collaborator or not
+	user := commentEvent.Sender.Login
+	comment := *commentEvent.Comment.Body
+	//split the assignees and operation to be performed.
+	substrings := strings.Split(comment, "@")
 
-	// https://github.com/islinwb/test/pull/1
-	prID := strings.SplitAfter(prc.Issue.PullRequestLinks.GetHTMLURL(), "github.com/")[1]
-	// https://github.com/islinwb/test/pull/1.patch
-	// From <commit ID> MON ...
-	patchURL := prc.Issue.PullRequestLinks.GetPatchURL()
-	resp, err := http.Get(patchURL)
+	assign, _, err := client.Repositories.IsCollaborator(ctx, owner, *commentEvent.Repo.Name, *user)
 	if err != nil {
-		fmt.Println(err)
-	}
+		glog.Fatalf("Not the collaborator: %v", err)
 
-	resp1, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		fmt.Println(err)
 	}
+	reg := regexp.MustCompile("(?mi)^/(un)?assign(( @?[-\\w]+?)*)\\s*$")
+	comm := reg.MatchString(comment)
+	listOfAssignees := make([]string, 0)
 
-	patchDetail := string(resp1)
-	reg := regexp.MustCompile(`From [A-Za-z0-9]{40}`)
-	commitIDstr := reg.FindString(patchDetail)
-	commitID := strings.TrimPrefix(commitIDstr, "From ")
-
-	var info map[string]string
-	info["PR_ID"] = prID
-	info["Commit_ID"] = commitID
-
-	if labelReg.MatchString(comment) {
-		labelSlice := strings.Split(comment, " ")
-		if len(labelSlice) > 0 {
+	for i, assignees := range substrings {
+		if i == 0 {
+			continue
 		}
+		listOfAssignees = append(listOfAssignees, assignees)
 	}
+	operation := strings.Trim(substrings[0], " ")
 
-	if retestReg.MatchString(comment) {
-		// "/retest"
-		s.SendToCI(info)
-	} else if testReg.MatchString(comment) {
-		// TODO: trigger particular job(s)
-		s.SendToCircleCI(body)
+	if assign {
+		if comm == true {
+			if operation == Assign {
+				issue, _, err := client.Issues.AddAssignees(ctx, *commentEvent.Repo.Owner.Login, *commentEvent.Repo.Name, *commentEvent.Issue.Number, listOfAssignees)
+				if err != nil {
+					glog.Fatalf("Cannot Add Assignees: %v", err)
+				}
+				if issue != nil {
+					glog.Info("Assignee added successfully")
+				}
+			} else if operation == Unassign {
+				_, _, err := client.Issues.RemoveAssignees(ctx, *commentEvent.Repo.Owner.Login, *commentEvent.Repo.Name, *commentEvent.Issue.Number, listOfAssignees)
+				if err != nil {
+					glog.Fatalf("Cannot remove Assignees: %v", err)
+
+				}
+				glog.Info("removed assignee!! ")
```

Comment:
remove -> Remove & remove unneeded white spaces.

Can we give more detailed information on whom are unassigned?